forms/HeaderField.php

```diff
@@ -17,31 +17,17 @@ class HeaderField extends DatalessField {
 	 */
 	protected $headingLevel = 2;
 
+	protected $schemaDataType = self::SCHEMA_DATA_TYPE_STRUCTURAL;
+
+	protected $schemaComponent = 'HeaderField';
+
 	/**
 	 * @param string $name
-	 * @param null|string $title
+	 * @param string $title
 	 * @param int $headingLevel
 	 */
-	public function __construct($name, $title = null, $headingLevel = 2) {
```

Comment:
I've checked that there are no core locations where this legacy behaviour was necessary.

It's acceptable for us to make breaking changes it it helps us better support the code in the future. In this case, removing the complex buggy support for legacy constructor is simpler than writing additional fixes to support numeric titles (which this code was crashing on).